storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4189,8 +4210,32 @@ void ha_rocksdb::load_auto_incr_value() {
   const bool save_keyread_only = m_keyread_only;
   m_keyread_only = true;
 
-  if (!index_last(table->record[0]))
-    update_auto_incr_val();
+  dict_manager.lock();
+  auto scf = dict_manager.get_system_cf();
+  uchar key_buf[Rdb_key_def::INDEX_NUMBER_SIZE * 3] = {0};
+
+  dict_manager.get_key_from_index_id((uchar *)key_buf,
+                                     m_pk_descr->get_gl_index_id());
+
+  auto k = rocksdb::Slice((char *)key_buf, sizeof(key_buf));
+  dict_manager.unlock();
+  std::string val;
+  tx->get(scf, k, &val);
+  longlong new_val =
+      rdb_netbuf_to_uint64(reinterpret_cast<const uchar *>(val.c_str()));
+    
+  if (new_val != std::numeric_limits<longlong>::max()) {
+    new_val++;
+  }
+
+  
+  longlong auto_incr_val = m_tbl_def->m_auto_incr_val;
+  while (auto_incr_val < new_val &&
+         !m_tbl_def->m_auto_incr_val.compare_exchange_weak(auto_incr_val,
+                                                           new_val)) {
+    // Do nothing - just loop until auto_incr_val is >= new_val or
+    // we successfully set it
+  }
```

Comment:
you could extract the test & set loop into a helper method that could be also called later in `ha_rocksdb::update_pk()`